### PR TITLE
Users can select tiles to make moves.

### PR DESCRIPTION
**Pull request by danielduffield**
Created an on click event that allows individual tiles to be selected and swapped with other tiles. Clicking an already selected tile will deselect it.

### CODE REVIEW

**Comment by danielduffield** [comment]
screen shot 2017-06-13 at 11 56 00 pm

File: main.js [hunk 1]
```diff
@@ -3,8 +3,11 @@ function generateTiles() {
   for (let i = 0; i < 8; i++) {
     for (let j = 0; j < 8; j++) {
       var tile = {}
+      tile.originRow = i
```
Comment (thebearingedge) [inline main.js]: Favor inline properties:
```js
var tile = {
  originRow: i,
  originCol: j,
  // ...etc
}

File: main.js [hunk 2]
```diff
@@ -28,6 +31,7 @@ function generateBoard(tiles) {
 
 function renderBoard(board) {
   var $board = document.createElement('div')
+  $board.setAttribute('id', 'board-render')
```
Comment (thebearingedge) [inline main.js]: If you are going to use `let`, be consistent. You shouldn't need `var` anymore.

File: main.js [hunk 3]
```diff
@@ -69,4 +108,16 @@ var startGame = function(event) {
   $container.removeChild($start)
 }
 
+var selectTile = function(event) {
+  if (!(event.target.classList[1] === 'hidden-tile') && event.target.classList[0] === 'board-tile') {
+    console.log(event.target)
```
Comment (thebearingedge) [inline main.js]: Remove all console logs before turning in code.

File: main.js [hunk 4]
```diff
@@ -69,4 +108,16 @@ var startGame = function(event) {
...
+    var $current = board[event.target.id[4]][event.target.id[13]]
+    $current.isSelected = !$current.isSelected
+    console.log($current)
+    swapTiles(board)
```
Comment (thebearingedge) [inline main.js]: Instead, try to return a new `board`.

File: main.js [hunk 5]
```diff
@@ -69,4 +108,16 @@ var startGame = function(event) {
...
+    updateBoard(board)
```
Comment (thebearingedge) [inline main.js]: Then, replace the old DOM `board` with the new DOM `board`.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**